doc/iterations/2024.5/shapes/add_support_for_representation_migration.adoc

```diff
@@ -0,0 +1,41 @@
+= Add the support for representations migration
+
+== Context
+
+Specifiers want to upgrade the version of Sirius Web without losing the representations they created with an older version.
+
+== Key Result
+
+Representations contributed with a version n of Sirius Web must be available with any version>n of Sirius Web. 
+
+Specifiers must also be able to contribute on how to handle evolutions on their own representations to guarantee compatibility between several versions.
+
+== Solution
+
+Whereas the models are deserialized/serialized by EMF, the representations use `Jackson` library to convert POJO into JSON.
+
+We can plug in modules into Jackson to give indications on how to serialize/deserialize JSON to/from Java.
+
+Specifier will be able to contribute several `IMigrationParticipant` that will be available in the implementation of a Jackson module.
```

Comment:
It can't be the same IMigrationParticipant as in the EMF Part, so I'll rename those to `IRepresentationMigrationParticipant`